process_report/process_report.py

```diff
@@ -333,9 +346,23 @@ def apply_credits_new_pi(dataframe, old_pi_file):
                 if remaining_credit == 0:
                     break
 
+    dump_old_pis(old_pi_file, old_pi_dict)
+
     return dataframe
 
 
+def fetch_s3_old_pi_file():
+    local_name = "PI.csv"
+    invoice_bucket = get_invoice_bucket()
+    invoice_bucket.download_file("PIs/PI.csv", local_name)
+    return local_name
+
+
+def upload_to_s3_old_pi_file(old_pi_file):
+    invoice_bucket = get_invoice_bucket()
+    invoice_bucket.upload_file(old_pi_file, "PIs/PI.csv")
```

Comment:
You're using the literal value `"PIs/PI.csv"` in two different functions. This should be a constant/global instead.

Reply:
For some of your suggested changes which are not directly related to the cronjob (i.e for this, and for the Dockerfile), they do relate to PRs that have closed previously. Should I open them and changes to those PRs, or what is the right way to handle this?

Reply:
If you don't think it's appropriate to address a comment in this PR, then it's fine to fix them in a future PR. A common workflow is to open an issue with the comment, and include a link to the issue in a response to the comment (and then at some point in the future, open a pull request that closes that issue).

Reply:
Should the fix to the action file, Dockerfile, and this line of code that you highlighted, be placed in seperate PRs, or would it be fine for them to be seperate commits bundled in one PR?

Reply:
I think it's fine to roll up a collection of small maintenance changes into a single pull request. :+1: for using separate commits for unrelated changes.